File: GGAndroid_3.0/core/support/rxjava/src/main/java/io/ganguo/rx/ActivityResult.java
package io.ganguo.rx;

import android.app.Activity;
import android.content.Intent;

/**
 * ActivityResult Container
 * Created by Roger on 8/10/16.
 */
public class ActivityResult {

    private final int requestCode;
    private final int resultCode;
    private final Intent data;

    public ActivityResult(Integer requestCode, Integer resultCode, Intent data) {
        this.requestCode = requestCode;
        this.resultCode = resultCode;
        this.data = data;
    }

    public boolean isResultOK() {
        return resultCode == Activity.RESULT_OK;
    }

    public int getRequestCode() {
        return requestCode;
    }

    public int getResultCode() {
        return resultCode;
    }

    public Intent getData() {
        return data;
    }

    @Override
    public String toString() {
        return "ActivityResult{" +
                "requestCode=" + requestCode +
                ", resultCode=" + resultCode +
                ", data=" + data +
                '}';
    }
}

File: GGAndroid_4.0/core/support/support-rx-activity-result/src/main/java/io/ganguo/rxresult/ActivityResult.kt
package io.ganguo.rxresult

import android.app.Activity
import android.content.Intent


/**
 * <pre>
 *     author : leo
 *     time   : 2020/05/29
 *     desc   : ActivityResult Container
 * </pre>
 */
class ActivityResult(val requestCode: Int, val resultCode: Int, val data: Intent?) {
    val isResultOK: Boolean
        get() = resultCode == Activity.RESULT_OK

    override fun toString(): String {
        return "ActivityResult{" +
                "requestCode=" + requestCode +
                ", resultCode=" + resultCode +
                ", data=" + data +
                '}'
    }

}